pkg/actuators/machine/reconciler.go

```diff
@@ -187,7 +187,30 @@ func (r *Reconciler) update() error {
 
 	klog.Infof("Updated machine %s", r.machine.Name)
 
-	r.machineScope.setProviderStatus(newestInstance, conditionSuccess())
+	vpc, err := r.awsClient.DescribeVpcs(&ec2.DescribeVpcsInput{
```

Comment:
can we wrap all this up into a function e.g `getCustomDNS` that we can unit test?